### PR TITLE
Feat/db place

### LINKED ISSUE #2179: It should be not possible to change welcome chat type from place to private

### PR DESCRIPTION
**Pull request by crui3er**
Persist place data to places table
Fixes #2179 It should be not possible to change welcome chat type from place to private

### CODE REVIEW

File: src/dotnet/Chat.Service/Places.cs [hunk 1]
```diff
@@ -95,19 +107,50 @@ public virtual async Task<ApiArray<AuthorId>> ListOwnerIds(Session session, Plac
...
+            chatDiff = !placeChange.Remove ? Change.Update(ToChatDiff(placeChange.Update.Value)) : Change.Remove<ChatDiff>();
+        }
+
+        place = await Commander.Call(changePlaceCommand, cancellationToken).ConfigureAwait(false);
+        rootChatId = rootChatId.Or(place.Id.ToRootChatId());
+
+        // Since place root chat is still used for getting place rules and place members we should synchronously update it.
+        // Should we use backend command here?
+        var chatChangeCommand = new Chats_Change(session, rootChatId, chatExpectedVersion, chatDiff);
+        placeRootChat = await Commander.Call(chatChangeCommand, true, cancellationToken).ConfigureAwait(false);
```
Comment (alexis-kochetov) [inline src/dotnet/Chat.Service/Places.cs]: Why do you use outermost command there?

Comment (alexis-kochetov) [reply src/dotnet/Chat.Service/Places.cs]: Usually we prefer backend command calls

File: src/dotnet/Chat.Service/Places.cs [hunk 2]
```diff
@@ -205,6 +226,25 @@ private static ChatDiff ToChatDiff(PlaceDiff placeDiff)
...
         };
 
+    private async Task RemovePlaceChats(Session session, PlaceId placeId, CancellationToken cancellationToken)
+    {
+        var contacts = await Contacts.ListIds(session, placeId, cancellationToken).ConfigureAwait(false);
+        foreach (var contact in contacts) {
+            var chatId = contact.ChatId;
+            if (chatId.IsPeerChat(out _))
+                continue; // Ignore peer chats
+            var chat = await Chats.Get(session, chatId, cancellationToken).ConfigureAwait(false);
```
Comment (alexis-kochetov) [inline src/dotnet/Chat.Service/Places.cs]: Please add `newline` after `continue`

File: src/dotnet/Chat.Service/Places.cs (hunk 1 above)
Comment (alexis-kochetov) [reply src/dotnet/Chat.Service/Places.cs]: Probably we should move this call to the Places backend so it can incapsulate all place-chat sync logic

**Review by alexis-kochetov — CHANGES_REQUESTED** [review]

File: src/dotnet/Chat.Service/Places.cs [hunk 3]
```diff
@@ -153,30 +196,8 @@ public virtual async Task OnPromoteToOwner(Places_PromoteToOwner command, Cancel
...
-                });
-                await Commander.Call(resetChatTagCommand, true, cancellationToken).ConfigureAwait(false);
-            }
-            var deleteChatCommand = new Chats_Change(session, chatId, null, new Change<ChatDiff> { Remove = true });
-            await Commander.Call(deleteChatCommand, true, cancellationToken).ConfigureAwait(false);
-        }
-
-        var deleteRootChatCommand = new Chats_Change(session, place.Id.ToRootChatId(), null, new Change<ChatDiff> { Remove = true });
-        await Commander.Call(deleteRootChatCommand, true, cancellationToken).ConfigureAwait(false);
+        var placeChangeCommand = new Places_Change(command.Session, command.PlaceId, null, Change.Remove<PlaceDiff>());
```
Comment (frolyo) [inline src/dotnet/Chat.Service/Places.cs:176]: if Places_Delete command for backward compatibility only, then maybe we need to mark it properly

**Review by frolyo — APPROVED** [review]

File: src/dotnet/Chat.Service/Places.cs (hunk 1 above)
Comment (crui3er) [reply src/dotnet/Chat.Service/Places.cs]: Reworked

File: src/dotnet/Chat.Service/Places.cs (hunk 3 above)
Comment (crui3er) [reply src/dotnet/Chat.Service/Places.cs:176]: Done

**Review by alexis-kochetov — APPROVED** [review]

**Pull request merged**